src/modules/member/components/Member.js

```diff
@@ -0,0 +1,229 @@
+            )
+        }
+
+        if(this.state.isDeleted){
+            return(
+                <div>
+                    <h1>Member is deleted!</h1>
+                    <Link to="/member">Go back to overview</Link>
+                </div>
+            )
+        }
+
+        if (this.state.isEditing) {
+            return (
+                <div>
+                    {this.renderEdit()}
+                </div>
+            )
+        } else {
+            return (
+                <div>{this.renderNormal()}</div>
+            )
+        }
+    }
+
+    handleNameChange(e){
+        this.setState({name: e.target.value})
+    }
+    handleEmailChange(e){
+        this.setState({email: e.target.value})
+    }
+    handleLifetimeChange(e){
+        this.setState({lifetime: e.target.checked})
+    }
+    handleHonoraryChange(e){
+        this.setState({honorary: e.target.checked})
+    }
+
+
+
+    renderEdit() {
+        return(
+            <div>
+
+                <button type="button" className="btn btn-default" onClick={this.abortEdit}>Abort</button>
+                <button type="button" className="btn btn-default" onClick={this.saveEdit}>Save</button>
+                <div className="panel-body">
+                    <form onsubmit={this.saveEdit}>
+                        <div className="row">
+                            <div className="form-group">
+                                <label htmlFor="name">Name</label>
+                                <input type="'text" name="'name" value={this.state.name}
+                                       onChange={this.handleNameChange} className="form-control"/>
+                                <label htmlFor="email">Email:</label>
+                                <input type="text" name='email' value={this.state.email}
+                                       onChange={this.handleEmailChange} className="form-control"/>
+                                <label htmlFor="lifetime">Lifetime</label>
+                                <input type="checkbox" name="lifetime" checked={this.state.lifetime}
+                                       onChange={this.handleLifetimeChange}/>
+                                <label htmlFor="honorary">Honorary</label>
+                                <input type="checkbox" name="honorary" checked={this.state.honorary}
+                                       onChange={this.handleLifetimeChange}/>
+                            </div>
+                        </div>
+                    </form>
+                </div>
+            </div>
+        )
+    }
+
+    abortEdit(e){
+        e.preventDefault();
+        this.setState({
+            isEditing : false
+        })
+    }
+
+    saveEdit(e){
+        e.preventDefault()
+        MemberService.updateMember(this.state.id, this.state.name, this.state.email,
+            this.state.lifetime, this.state.honorary).then(result => {
+            actions.updateMember(this.state.id)
+            this.setState({
```

Comment:
Hva med bare `this.setState(result)` ? Ser dog en mulig fordel slik det er nå at man eksplisitt ser hva som settes